File: CreatorNote/Supabase/DTOs/GeneralNoteDTO.swift
import Foundation

struct GeneralNoteDTO: Codable, Identifiable, Sendable {
    let id: UUID
    var workspaceId: UUID
    var title: String
    var attributedContent: Data?
    var plainContent: String
    var tags: [String]
    var isPinned: Bool
    var createdBy: UUID?
    var createdAt: Date
    var updatedAt: Date

    enum CodingKeys: String, CodingKey {
        case id, title, tags
        case workspaceId = "workspace_id"
        case attributedContent = "attributed_content"
        case plainContent = "plain_content"
        case isPinned = "is_pinned"
        case createdBy = "created_by"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

File: android/app/src/main/java/com/soohoon/influe/data/model/GeneralNoteDTO.kt
package com.soohoon.influe.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class GeneralNoteDTO(
    val id: String,
    @SerialName("workspace_id") val workspaceId: String,
    val title: String = "",
    @SerialName("attributed_content") val attributedContent: String? = null,
    @SerialName("plain_content") val plainContent: String = "",
    val tags: List<String> = emptyList(),
    @SerialName("is_pinned") val isPinned: Boolean = false,
    @SerialName("created_by") val createdBy: String? = null,
    @SerialName("created_at") val createdAt: String = "",
    @SerialName("updated_at") val updatedAt: String = ""
)

@Serializable
data class GeneralNoteInsert(
    @SerialName("workspace_id") val workspaceId: String,
    @SerialName("created_by") val createdBy: String,
    val title: String = "",
    @SerialName("plain_content") val plainContent: String = "",
    @SerialName("attributed_content") val attributedContent: String? = null,
    val tags: List<String> = emptyList()
)
